src/amo/api/collections.js

```diff
@@ -285,3 +287,79 @@ export const createCollection = ({
     _validateLocalizedString,
   });
 };
+
+type ModifyCollectionAddonBaseParams = {|
+  addonId: number,
+  api: ApiStateType,
+  collectionSlug: string,
+  user: string,
+  _modifyCollectionAddon?: (any) => Promise<void>,
+|};
+
+export type CreateCollectionAddonParams = {|
+  ...ModifyCollectionAddonBaseParams,
+  notes?: CollectionAddonNotes,
+|};
+
+export type UpdateCollectionAddonParams = {|
+  ...ModifyCollectionAddonBaseParams,
+  notes: CollectionAddonNotes,
+|};
+
+type ModifyCollectionAddonParams =
+  | {| action: 'create', ...CreateCollectionAddonParams |}
+  | {| action: 'update', ...UpdateCollectionAddonParams |};
+
+export const modifyCollectionAddon = (
+  params: ModifyCollectionAddonParams,
+): Promise<void> => {
+  const { action, api, user, collectionSlug, addonId } = params;
+
+  invariant(action, 'The action parameter is required');
+  invariant(addonId, 'The addonId parameter is required');
+  invariant(api, 'The api parameter is required');
+  invariant(collectionSlug, 'The collectionSlug parameter is required');
+  invariant(user, 'The user parameter is required');
+
+  let method = 'POST';
+  const body = { addon: addonId, notes: params.notes || null };
+  let endpoint =
+    `accounts/account/${user}/collections/${collectionSlug}/addons`;
+
+  if (action === 'update') {
+    if (typeof params.notes === 'undefined') {
```

Comment:
Why not just `params.notes === 'undefined'`?

Reply:
Oops, I of course meant `param.notes === undefined`.

Reply:
Oh, heh, it is an old habit from IE6 days where referencing an undefined name would throw. I can change it.